### PR TITLE
recent activity tab

### LINKED ISSUE #1434: Country Home/Recent activity

### PR DESCRIPTION
**Pull request by rfontanarosa**
closes #1434

Schermata 2022-08-24 alle 16 26 17

Schermata 2022-08-24 alle 15 21 18

### CODE REVIEW

File: server/api/assessment/getActivities.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Response } from 'express'
+
+import { CycleDataRequest } from '@meta/api/request'
+
+import { AssessmentController } from '@server/controller/assessment'
+import { CycleDataController } from '@server/controller/cycleData'
+import Requests from '@server/utils/requests'
+
+export const getActivities = async (req: CycleDataRequest, res: Response) => {
```
Comment (minotogna) [inline server/api/assessment/getActivities.ts:9]: this goes under server/api/cycleData

File: server/api/cycleData/table/persistNodeValues.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export const persistNodeValues = async (req: CycleDataRequest<never, NodesBody>,
           countryIso,
           assessment,
           cycle,
+          section,
```
Comment (minotogna) [inline server/api/cycleData/table/persistNodeValues.ts:27]: pass the parameter as sectionName. this should be ```sectionName:section```
in CycleDataController.persistNodeValue and all related method use sectionName

File: server/repository/assessment/activityLog/getCycleDataActivities.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        a.*,
+        to_jsonb(u.*) - 'profile_picture_file' - 'profile_picture_filename' as user  
+      from (
+        select
+          user_id, message, section, target, time,
+          rank() OVER (PARTITION BY user_id, message, section ORDER BY time DESC) as rank
+        from ${schema}.activity_log a
+        where a.country_iso = $1
+          and a.cycle_uuid = $2
+          and a.message not in ('deleteComment', 'nodeValueCalculatedUpdate', 'updateCountry')
```
Comment (minotogna) [inline server/repository/assessment/activityLog/getCycleDataActivities.ts:28]: use affermative condition. message in (...)

**Review by minotogna — CHANGES_REQUESTED** [review]

File: common/api/endpoint/ApiEndPoint.ts [hunk 4]
```diff
@@ -23,6 +23,7 @@ export const ApiEndPoint = {
       nodes: () => apiPath('cycleData', 'table', 'nodes'),
       tableData: () => apiPath('cycleData', 'table', 'tableData'),
     },
+    activities: () => apiPath('cycleData', 'activities'), // is it correct?
```
Comment (sorja) [inline common/api/endpoint/ApiEndPoint.ts:26]: it is correct (remove comment)

File: client/pages/AssessmentHome/FraHome/RecentActivity/RecentActivityItem/RecentActivityItem.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+.landing__activity-time {
+  color: $text-mute;
+  font-size: $font-s;
+  margin-left: auto;
+  white-space: nowrap;
+}
+
+.landing__activity-empty {
+  color: $text-disabled;
+  padding: 48px;
```
Comment (sorja) [inline client/pages/AssessmentHome/FraHome/RecentActivity/RecentActivityItem/RecentActivityItem.scss:47]: should we use `$spacing-xl: 48px;`?

File: server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateDependantNodes.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ import { RowRepository } from '@server/repository/assessment/row'
 import { calculateNode } from './calculateNode'
 
 export const calculateDependantNodes = async (props: Props, client: BaseProtocol): Promise<NodeUpdates> => {
-  const { assessment, cycle, countryIso, tableName, variableName, colName, user } = props
+  const { assessment, cycle, countryIso, section, tableName, variableName, colName, user } = props
```
Comment (sorja) [inline server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateDependantNodes.ts:11]: sectionName?

File: server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateNode.ts [hunk 7]
```diff
@@ -31,6 +31,7 @@ export const calculateNode = async (
       countryIso,
       assessment,
       cycle,
+      section,
```
Comment (sorja) [inline server/controller/cycleData/persistNodeValue/calculateDependantNodes/calculateNode.ts:34]: sectionName?

File: server/controller/cycleData/persistNodeValue/persistNode/persistNode.ts [hunk 8]
```diff
@@ -10,14 +10,14 @@ export const persistNode = async (
...
-  const { assessment, countryIso, activityLogMessage, user, cycle } = props
+  const { assessment, countryIso, activityLogMessage, user, cycle, section } = props
   const node: Node = await NodeRepository.getOneOrNone(props, client)
 
   const nodeUpdated = await (node ? NodeRepository.update(props, client) : NodeRepository.create(props, client))
   const activityLog: ActivityLog<Node> = {
     countryIso,
     message: activityLogMessage ?? ActivityLogMessage.nodeValueUpdate,
-    section: 'node',
+    section,
```
Comment (sorja) [inline server/controller/cycleData/persistNodeValue/persistNode/persistNode.ts:20]: sectionName?

Comment (sorja) [reply server/controller/cycleData/persistNodeValue/persistNode/persistNode.ts:20]: section: sectionName for now maybe

File: server/controller/cycleData/persistNodeValue/props.ts [hunk 9]
```diff
@@ -7,5 +7,6 @@ export type Props = {
   countryIso: CountryIso
   assessment: Assessment
   cycle: Cycle
+  section?: string
```
Comment (sorja) [inline server/controller/cycleData/persistNodeValue/props.ts:10]: sectionName

**Review by sorja — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**